source/common/network/connection_impl.cc

```diff
@@ -604,6 +604,7 @@ void ConnectionImpl::onWriteBufferHighWatermark() {
 void ConnectionImpl::setFailureReason(absl::string_view failure_reason) {
   if (!transport_socket_->failureReason().empty()) {
     failure_reason_ = absl::StrCat(failure_reason, ". ", transport_socket_->failureReason());
+    stream_info_.setDownstreamTransportFailureReason(failure_reason_);
```

Comment:
Hi @kyessenov, can you please let me know if this is the right place to set the attribute as per your previous suggestion?
https://github.com/envoyproxy/envoy/pull/29180/files#r1317709061

Reply:
I think this is not needed. The failure reason is already set in `ActiveTcpConnection::onEvent` for `close` events.